crwatcher/watcher.go

```diff
@@ -114,15 +115,24 @@ func (cw *CRWatcher) setupHandler(con ResourceController) {
 	cw.handler = cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			r := obj.(*unstructured.Unstructured)
+			if cw.Config.OptInKey != "" && !cw.hasOptInKey(r) {
+				return
+			}
 			con.ResourceAdded(r)
 		},
 		DeleteFunc: func(obj interface{}) {
 			r := obj.(*unstructured.Unstructured)
+			if cw.Config.OptInKey != "" && !cw.hasOptInKey(r) {
+				return
+			}
 			con.ResourceDeleted(r)
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			oldR := oldObj.(*unstructured.Unstructured)
 			newR := newObj.(*unstructured.Unstructured)
+			if cw.Config.OptInKey != "" && (!cw.hasOptInKey(oldR) && !cw.hasOptInKey(newR)) {
```

Comment:
We should only check if the `newR` has the optIn key here, I think. 

An interesting scenario though is what happens where the old has the optIn key and the new record doesn't. Should that trigger a delete?

Reply:
We decided that optIn on the old, but not on the new should perform a Delete. All other cases should act as expected

No && No => No-Op
OptIn && OptIn => Update
No && OptIn => Update

This will be documented in an events doc.